main.py

```
@@ -118,14 +122,45 @@ async def register_user(sid, data):
+    try:
+        expires_str = admin_session.get("expires_at", "")
+        if expires_str:
+            # Parse ISO format datetime
+            expires_at = datetime.fromisoformat(expires_str.replace("Z", "+00:00"))
+            now = datetime.now(timezone.utc)
+            if expires_at < now:
+                logger.warning(f"🚫 Expired token used: {sid}")
+                del authenticated_admins[token]
+                await sio.emit('auth_error', {"error": "Session expired. Please login again."}, room=sid)
+                return
+    except Exception as e:
```

Ruff (0.15.4)

[warning] Do not catch blind exception: `Exception`

(BLE001)